File: schemas/multi-language-i18n/drizzle/translation_reviews.ts
// translation_reviews: Review actions taken on translations.
// See README.md for full design rationale.

import { pgEnum, pgTable, uuid, text, timestamp, index } from "drizzle-orm/pg-core";
import { users } from "../../auth-rbac/drizzle/users";

export const translationReviewActionEnum = pgEnum("translation_review_action", [
  "approve",
  "reject",
  "request_changes",
]);

export const translationReviews = pgTable(
  "translation_reviews",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    translationType: text("translation_type").notNull(),
    translationId: uuid("translation_id").notNull(),
    reviewerId: uuid("reviewer_id")
      .notNull()
      .references(() => users.id, { onDelete: "cascade" }),
    action: translationReviewActionEnum("action").notNull(),
    comment: text("comment"),
    createdAt: timestamp("created_at", { withTimezone: true })
      .notNull()
      .defaultNow(),
  },
  (table) => [
    index("idx_translation_reviews_translation_type_translation_id").on(
      table.translationType,
      table.translationId
    ),
    index("idx_translation_reviews_reviewer_id").on(table.reviewerId),
  ]
);

File: schemas/multi-language-i18n/spacetimedb/translation_reviews.rs
// translation_reviews: Append-only log of translation review actions.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

/// Approve = reviewer approves the translation.
/// Reject = reviewer rejects the translation.
/// RequestChanges = reviewer requests modifications.
// type: String
#[derive(SpacetimeType, Clone)]
pub enum TranslationReviewAction {
    Approve,
    Reject,
    RequestChanges,
}

// Composite index: (translation_type, translation_id) — all reviews for a translation

#[spacetimedb::table(name = translation_reviews, public)]
pub struct TranslationReview {
    #[primary_key]
    pub id: String, // UUID

    #[index(btree)]
    pub translation_type: String,

    pub translation_id: String,

    #[index(btree)]
    pub reviewer_id: String, // FK → users.id (cascade delete)

    pub action: TranslationReviewAction,
    pub comment: Option<String>,
    pub created_at: Timestamp,
}
